### PR TITLE
Update Legacy Demonstration with Refactored Code

### LINKED ISSUE #1610: Update Legacy Demonstration with Refactored Code
With the newly refactored code in place, the custom-scripts team has decided to update our team's demonstration from two years ago so that it is usable with our new codebase. This includes renaming our include modules and updating both the name of function calls and inputting the correct parameters.

This issue is completed when we are able to run the demonstration.c such that a game of chiventure is loaded and playable.

### PR DESCRIPTION
**Pull request by tatmaca**
Pull request to resolve Issue #1610 : 

"With the newly refactored code in place, the custom-scripts team has decided to update our team's demonstration from two years ago so that it is usable with our new codebase. This includes renaming our include modules and updating both the name of function calls and inputting the correct parameters.

This issue is completed when we are able to run the demonstration.c such that a game of chiventure is loaded and playable."

### CODE REVIEW

**Review by mxu24 — APPROVED** [review]

**Review by mxu24 — APPROVED** [review]
All code and files appear appropriate, demo runs accurately

**Review by ecoble — COMMENTED** [review]
Is this a duplication of PR #1586?

**Comment by tatmaca** [comment]
> 

@ecoble This pull request updates the demonstration.c file from two years ago so that it runs with the new refactored code! PR #1586 creates a new demo from scratch which plays around with more in-depth lua scripting! I think what may have happened is that when we created the branch seen in PR #1586, we merged it with one of our other branches awaiting a pull request because the refactored code wasn't yet available in the dev branch. In that merge, it looks like some code from this branch was also copied over. Rest assured however, the PRs are unique and address their own respective issues [🙂](https://emojipedia.org/slightly-smiling-face/)👍 This PR is us reworking the original demo so that it works with the refactored code, and the other PR is us experimenting with Lua and creating a new demo from scratch!

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[submodule "include/custom-scripts/--foorce"]
```
Comment (ecoble) [inline .gitmodules]: Please remove submodules

**Review by ecoble — CHANGES_REQUESTED** [review]
The header files in tests/custom-scripts seem like a duplication of the header files that already were merged in include, why not reference those instead of creating new header files?

**Review by ecoble — APPROVED** [review]
Nice job cleaning things up!

**Comment by jackjackkimkim** [comment]
We were able to make updates to the legacy demonstrations to function correctly with the refactored code. The old demonstration can now make with the new refactored code design.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great job updating the code and responded to changes on this PR!

**Pull request merged**